pkg/kfapp/kustomize/kustomize.go

```diff
@@ -1283,6 +1322,86 @@ func GenerateKustomizationFile(kfDef *kfconfig.KfConfig, root string,
 	return kustomizationPathErr
 }
 
+// AddGlobalTransformers adds globaly defined transformers to the kustomization
+func AddGlobalTransformers(kfDef *kfconfig.KfConfig, kustomization *types.Kustomization, compDir string) *kfapisv3.KfError {
+	transformersDir := filepath.Join(compDir, transformersDirName)
+	if len(kfDef.Spec.Global.Transformers) > 0 {
+		if err := os.MkdirAll(transformersDir, os.ModePerm); err != nil {
+			err := fmt.Errorf("could not create directory %v", transformersDir)
+			log.Errorf("%v", err)
+			return &kfapisv3.KfError{
+				Code:    int(kfapisv3.INTERNAL_ERROR),
+				Message: err.Error(),
+			}
+		}
+	}
+
+	for _, t := range kfDef.Spec.Global.Transformers {
+		repoCache, ok := kfDef.GetRepoCache(t.RepoRef.Name)
+		if !ok {
+			err := fmt.Errorf("could note get repo cache for repo %s", t.RepoRef.Name)
+			log.Errorf("%v", err)
+			return &kfapisv3.KfError{
+				Code:    int(kfapisv3.INTERNAL_ERROR),
+				Message: err.Error(),
+			}
+		}
+		transformetTargetFile := filepath.Join(transformersDir, filepath.Base(t.RepoRef.Path))
+		transformetTargetFileRelative := filepath.Join(transformersDirName, filepath.Base(t.RepoRef.Path))
+		if err := copy.Copy(filepath.Join(repoCache.LocalPath, t.RepoRef.Path), transformetTargetFile); err != nil {
+			return &kfapisv3.KfError{
+				Code:    int(kfapisv3.INTERNAL_ERROR),
+				Message: fmt.Sprintf("couldn't copy transformer %s to %s: %v", t.Name, transformetTargetFile, err),
+			}
+		}
+		add := true
+		for _, tx := range kustomization.Transformers {
+			if tx == transformetTargetFileRelative {
+				add = false
+				break
+			}
+		}
+
+		if add {
+			kustomization.Transformers = append(kustomization.Transformers, transformetTargetFileRelative)
+		}
+	}
+
+	return nil
+}
+
+func AddOdhDefaultTransformers(kustomization *types.Kustomization) error {
+	config, _ := rest.InClusterConfig()
+	corev1client, err := corev1.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	operatorNs , err := k8sutil.GetOperatorNamespace()
+	if err != nil{
+		return err
+	}
+	cfmap, err := corev1client.ConfigMaps(operatorNs).Get("odh-default-plugins", metav1.GetOptions{})
```

Comment:
Should the CM name be a constant at the top of the file to avoid having "magic constants" in the middle of the code?